pkg/pillar/hypervisor/kvm.go

```diff
@@ -1168,3 +1217,76 @@ func GetQmpExecutorSocket(domainName string) string {
 func getQmpListenerSocket(domainName string) string {
 	return filepath.Join(kvmStateDir, domainName, "listener.qmp")
 }
+
+func requestVtpmLaunch(id string, timeoutSeconds uint) (string, error) {
+	conn, err := net.Dial("unix", types.VtpmdCtrlSocket)
+	if err != nil {
+		return "", fmt.Errorf("failed to connect to vtpmd control socket: %w", err)
+	}
+	defer conn.Close()
+
+	pidPath := fmt.Sprintf(types.SwtpmPidPath, id)
+	sockPath := fmt.Sprintf(types.SwtpmCtrlSocketPath, id)
+
+	// Send the request to the vTPM control socket, ask it to launch a swtpm instance.
+	_, err = conn.Write([]byte(fmt.Sprintf("%s%s\n", vtpmlaunchPrefix, id)))
+	if err != nil {
+		return "", fmt.Errorf("failed to write to vtpmd control socket: %w", err)
+	}
+
+	// Loop and wait for SWTPM to start.
+	startTime := time.Now()
+	pid := 0
+	for {
+		if time.Since(startTime).Seconds() >= float64(timeoutSeconds) {
+			return "", fmt.Errorf("timeout reached while waiting for SWTPM to launch")
+		}
+
+		pid, err = utils.GetPidFromFile(pidPath)
+		if err == nil {
+			break
+		}
+
```

Comment:
I think it's better to call the `stillAlive` function here to be friendly to a watchdog.

Reply:
OK, these needs some work, propagating the domain manager info to kvm, I will change it.